lib/index.js

```diff
@@ -1,92 +1,117 @@
 import _ from 'lodash';
 
-export default class PiePlayer extends HTMLElement{
-
-  createdCallback(){
-    console.log('created');
+export default class PiePlayer extends HTMLElement {
+
+  createdCallback() {
+    this._pies = {};
+
+    this.addEventListener('pie.register', (event) => {
+      //stop the event from bubbling further
+      event.preventDefault();
+      event.stopImmediatePropagation();
+      let id = event.target.getAttribute('data-id');
+      if (this._pies[id]) {
+        throw new Error('a pie with that id (' + id + ') is already registered: ' + this._pies[id]);
+      }
+      this._pies[id] = event.target;
+      this._updateElement(event.target, id);
+    });
   }
-  
-  attachedCallback(){
-    console.log('attached');
-    let event = new CustomEvent('pie-player-ready', {bubbles: true});
+
+  attachedCallback() {
```

Comment:
can we do that as a separate issue so as not to delay this pr? i think there's a good chance that there'll be build issues, so would make sense to do them altogether